bundles/org.openhab.binding.energenie/src/main/java/org/openhab/binding/energenie/internal/handler/EnergeniePWMHandler.java

```diff
@@ -0,0 +1,213 @@
+                logger.trace("transformed state response = {}", slicedResponseTmp);
+                final String[] slicedResponse = slicedResponseTmp.split(";");
+                logger.trace("transformed state response = {} - {}", slicedResponse[0], slicedResponse[1]);
+                final double value;
+
+                try {
+                    if (Double.parseDouble(slicedResponse[0]) / 1 == Double.parseDouble(slicedResponse[0])) {
+                        value = Double.parseDouble(slicedResponse[0]) / divisor;
+                    } else {
+                        value = -1.0;
+                    }
+                    return QuantityType.valueOf(value, unit);
+
+                } catch (NumberFormatException e) {
+                    logger.debug("Could not Parse State", e);
+                    return UnDefType.UNDEF;
+                }
+            } else {
+                logger.trace("searchstring '{} not found", stateResponseSearch);
+                return UnDefType.UNDEF;
+            }
+        }
+    }
+
+    private final Logger logger = LoggerFactory.getLogger(EnergeniePWMHandler.class);
+
+    private static final int HTTP_TIMEOUT_MILLISECONDS = 6000;
+
+    private @Nullable EnergenieConfiguration config;
+
+    private String host = "";
+    private String password = "";
+    private int refreshInterval;
+
+    private @Nullable ScheduledFuture<?> refreshJob;
+
+    public EnergeniePWMHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            try {
+                getState();
+            } catch (Exception e) {
+                logger.debug("Exception during poll", e);
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        EnergenieConfiguration config = getConfigAs(EnergenieConfiguration.class);
+
+        this.config = config;
+
+        if (!config.host.isEmpty() && !config.password.isEmpty()) {
+            host = config.host;
+            password = config.password;
+            refreshInterval = EnergenieConfiguration.DEFAULT_REFRESH_INTERVAL;
+            logger.debug("Initializing EnergeniePWMHandler for Host '{}'", host);
+            updateStatus(ThingStatus.ONLINE);
+            onUpdate();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "Can not access device , IP-Address or password not set");
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("EnergeniePWMHandler disposed.");
+        final ScheduledFuture<?> refreshJob = this.refreshJob;
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            this.refreshJob = null;
+        }
+    }
+
+    public void getState() {
```

Comment:
You should make this synchronized to prevent threading issues between a RefreshCommand and a scheduled refresh.
```suggestion
    public synchronized void getState() {
```

Reply:
Done